src/Provider/FileProvider.php

```diff
@@ -351,6 +351,14 @@ protected function fixBinaryContent(MediaInterface $media)
             return;
         }
 
+        // if the binary content is a Symfony\Component\Finder\SplFileInfo => convert to a valid File
+        if ($media->getBinaryContent() instanceof \SplFileInfo) {
+            $binaryContent = new File($media->getBinaryContent()->getPathName());
+            $media->setBinaryContent($binaryContent);
+
+            return;
+        }
+
         // if the binary content is a filename => convert to a valid File
         if (!is_file($media->getBinaryContent())) {
```

Comment:
I have a custom file provider, where for some edge cases in this point I've added this check long time ago:
```
$content = $media->getBinaryContent();
if (is_string($content)) {
    // Replacing null bytes is required in order to avoid `E_WARNING` error
    // "PHP Warning: is_file() expects parameter 1 to be a valid path, string given"
    // @see: https://bugs.php.net/bug.php?id=76406&edit=3
    // @see: https://stackoverflow.com/a/22957831
    $sanitizedContent = str_replace("\0", '', $content);
    $isFile = PHP_MAXPATHLEN >= mb_strlen($sanitizedContent) && is_file($sanitizedContent);
}
```

Reply:
When can this happen?